lib/elixir/lib/list.ex

```diff
@@ -1,6 +1,16 @@
 defmodule List do
   @moduledoc """
   Functions that work on (linked) lists.
+  
+  Please note that most of the functions provided for lists--which implement 
+  the `Enumerable` protocol--are found in `Enum`.
+  
+  Additionally, the following functions for lists are found in `Kernel`:
+  
+    * `hd/1`
+    * `tl/1`
+    * `in/2` - equivalent to `Enum.member?/2`
+    * `length/1` - requires traversing the full list
```

Comment:
I don't think this is the place to mention that `length/1` requires traversing the whole list.